src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,44 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * The Deadline class contains the name, completion status
+ * and deadline of a task.
+ */
+public class Deadline extends Task {
+    protected LocalDate by;
```

Comment:
Would a more descriptive name for the date like "deadlineTiming" be a little clearer for readers? Calling task.getBy() might not be that obvious for first time readers.